migrations/__tests__/00004_notifications.spec.ts

```diff
@@ -0,0 +1,615 @@
+
+        return {
+          notification_types: await sql<
+            Array<NotificationTypesRow>
+          >`SELECT * FROM notification_types`,
+          notification_subscriptions: await sql<
+            Array<NotificationSubscriptionsRow>
+          >`SELECT * FROM notification_subscriptions`,
+          notification_subscription_notification_types: await sql<
+            Array<NotificationSubscriptionNotificationTypesRow>
+          >`SELECT * FROM notification_subscription_notification_types`,
+          notification_channels: await sql<
+            Array<NotificationChannelsRow>
+          >`SELECT * FROM notification_channels`,
+          notification_channel_configurations: await sql<
+            Array<NotificationChannelConfigurationsRow>
+          >`SELECT * FROM notification_channel_configurations`,
+        };
+      },
+    });
+
+    expect(result.after).toStrictEqual({
+      notification_types: [
+        {
+          id: expect.any(Number),
+          name: 'DELETED_MULTISIG_TRANSACTION',
+        },
+        {
+          id: expect.any(Number),
+          name: 'EXECUTED_MULTISIG_TRANSACTION',
+        },
+        {
+          id: expect.any(Number),
+          name: 'INCOMING_ETHER',
+        },
+        {
+          id: expect.any(Number),
+          name: 'INCOMING_TOKEN',
+        },
+        {
+          id: expect.any(Number),
+          name: 'MESSAGE_CREATED',
+        },
+        {
+          id: expect.any(Number),
+          name: 'MODULE_TRANSACTION',
+        },
+        {
+          id: expect.any(Number),
+          name: 'NEW_CONFIRMATION',
+        },
+        {
+          id: expect.any(Number),
+          name: 'MESSAGE_CONFIRMATION',
+        },
+        {
+          id: expect.any(Number),
+          name: 'OUTGOING_ETHER',
+        },
+        {
+          id: expect.any(Number),
+          name: 'OUTGOING_TOKEN',
+        },
+        {
+          id: expect.any(Number),
+          name: 'PENDING_MULTISIG_TRANSACTION',
+        },
+        {
+          id: expect.any(Number),
+          name: 'SAFE_CREATED',
+        },
+      ],
+      notification_subscriptions: [
+        {
+          id: 1,
+          chain_id: '1',
+          account_id: 1,
+          safe_address: '0x420',
+          created_at: expect.any(Date),
+          updated_at: expect.any(Date),
+        },
+        {
+          id: 2,
+          chain_id: '2',
+          account_id: 1,
+          safe_address: '0x420',
+          created_at: expect.any(Date),
+          updated_at: expect.any(Date),
+        },
+        {
+          id: 3,
+          chain_id: '3',
+          account_id: 1,
+          safe_address: '0x420',
+          created_at: expect.any(Date),
+          updated_at: expect.any(Date),
+        },
+      ],
+      notification_subscription_notification_types: [],
+      notification_channels: [],
+      // No configurations should exist
+      notification_channel_configurations: [],
+    });
+  });
```

Comment:
Nit: I'd include another test verifying that `notification_channel_configurations.device_type` has to be one of `('ANDROID', 'IOS', 'WEB')`.

Reply:
Added in bbf887220263d3371eb3bbe8021baa16e7b1470c.